Vagrantfile

```diff
@@ -9,6 +9,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     config.vm.network :forwarded_port, guest: 8282, host: 8282
     config.vm.network :forwarded_port, guest: 80, host: 8000
     config.vm.network :forwarded_port, guest: 4443, host: 4443
+    config.vm.network :forwarded_port, guest: 777, host: 777
```

Comment:
> there is something else that might work though I saw somewhere, setting ssl_verify_client to optional and then if in the jsonrpc location they don't have client-side verification, you 403 them

Yeah, that's what we did the last time we had to use Nginx for websockets where I work.  You set the actual SSL parameters in the ``server`` block but then say that the client cert is required in the ``location`` block.  This was years ago (Apache added the ability to proxy websocket traffic so we switched back to that), but I can try to dig up and/or reproduce the config we used if that would be helpful.